tmk_core/common/keyboard.c

```diff
@@ -130,10 +130,15 @@ static inline bool has_ghost_in_row(uint8_t row, matrix_row_t rowdata)
 #endif
 
 void disable_jtag(void) {
-// To use PORTF disable JTAG with writing JTD bit twice within four cycles.
-#if (defined(__AVR_AT90USB1286__) || defined(__AVR_AT90USB1287__) || defined(__AVR_ATmega32U4__))
+// To use PF4-7, disable JTAG by writing JTD bit twice within four cycles.
+#if (defined(__AVR_AT90USB646__)  || defined(__AVR_AT90USB647__)  || \
+     defined(__AVR_AT90USB1286__) || defined(__AVR_AT90USB1287__) || \
+     defined(__AVR_ATmega16U4__)  || defined(__AVR_ATmega32U4__))
     MCUCR |= _BV(JTD);
     MCUCR |= _BV(JTD);
+#elif defined(__AVR_ATmega32A__)
```

Comment:
Elsewhere, differences between MCUCR and MCUCSR are handled slightly differently.
https://github.com/qmk/qmk_firmware/blob/a8427447bc2204df5a21372e129fcfa5ff87e634/tmk_core/common/avr/bootloader.c#L68-L73

Reply:
I think it's better to do it this way (at least, in this instance), otherwise you need to wrap it in another ifdef that covers both cases. Neither of these registers exist on ARM 😉

Reply:
ARM has its own file for its implementation so that case would not apply.

I don't see why it would need to be any more than the code snippet above the block of code or at the top of the file. Having it done the same way in both cases adds some consistency to the code base and maybe a case for it to be put somewhere more generic.

Reply:
There's only one keyboard.c, and only one implementation of `disable_jtag()`...

Reply:
I should also note that not all AVRs have a JTAG interface. While your suggestion works when the bits needed are more ubiquitous (such as WDRF, most if not all AVRs have a watchdog timer), it will fail to compile on the ATmega328P.

Reply:
Sorry my mistake, too many tabs open. The code doesnt need to be more than
```c
void disable_jtag(void) {
// To use PF4-7, disable JTAG by writing JTD bit twice within four cycles.
#if (defined(__AVR_AT90USB646__)  || defined(__AVR_AT90USB647__)  || \
     defined(__AVR_AT90USB1286__) || defined(__AVR_AT90USB1287__) || \
     defined(__AVR_ATmega16U4__)  || defined(__AVR_ATmega32U4__) || \
     defined(__AVR_ATmega32A__))

 #if defined (MCUCSR) 
     #if defined (!MCUSR) 
         #define MCUSR MCUCSR
     #endif 
 #endif 

    MCUCR |= _BV(JTD);
    MCUCR |= _BV(JTD);
#endif

```

I will leave it as a loose approval, and see what some of the other Collaborators say.

Reply:
Honestly, that code seems a lot less clear as to why the aliasing is needed. Sure, it duplicates the implementation, but... it's two lines.

Reply:
I think I have a slight preference for @fauxpark's implementation. Yes, it's duplicated code, but I find it easier to follow than a bunch of nested `ifdef`s.

Caveat: I have no idea what JTAG is, and only know that it requires disabling so that its pins can be used for GPIO.

Reply:
@noroadsleft it's used for programming, and debugging, usually. 

I'm good with the code, as is. and yeah, I think it's a bit more readable as is.